src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -216,13 +217,44 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
             if (!foundBookmark) {
               // to capture updates to existing docs when maxPagesPerBatch is being used
               // adds the updated docs to the list while searching for the bookemarked Id from the prior batch
-              if (feedItem.getInt("_lsn") > currentContinuation.toInt) {
-                cfDocuments.add(feedItem)
-              }
-
-              if (feedItem.get("id") == lastProcessedIdBookmark) {
+              val lsn = feedItem.getInt("_lsn")
+              if (lsn > currentContinuation.toInt) {
+
+                if (shouldInferStreamSchema) {
+                  cfDocumentsIfLastProcessedDocumentWasUpdatedAgain.add(feedItem)
+                }
+                else {
+                  val streamDocument: Document = new Document()
+                  streamDocument.set("body", feedItem.toJson)
+                  streamDocument.set("id", feedItem.get("id"))
+                  streamDocument.set("_rid", feedItem.get("_rid"))
+                  streamDocument.set("_self", feedItem.get("_self"))
```

Comment:
why do we need to keep track of the system properties?
aren't these system generated?

Reply:
to allow reading them - keeping this as is here (same code has been used in other place couple of lines below)